### PR TITLE
Rbac issues

### PR DESCRIPTION
**Pull request by aliok**
JIRA: https://issues.jboss.org/browse/AEROGEAR-9570

Verification:
* Change the image in operator.yaml to this: aliok/mobile-developer-console-operator:jul10-2
* Follow the steps in README

cc @jhellar 

Note: MDC won't listen to other namespaces for any resources anymore. So, got rid of the `ClusterRole` and added some more stuff in the MDC role.

### CODE REVIEW

File: deploy/mobileclient_admin_role.yaml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+      - mobileclients
+    verbs:
+      - create
+      - delete
+      - get
+      - list
+      - patch
+      - update
+      - watch
+  - apiGroups:
```
Comment (aliok) [inline deploy/mobileclient_admin_role.yaml:31]: needed to have push and metrics related stuff as well, as the MDC users will need to work on those as well as mobileclients.

File: deploy/role.yaml [hunk 2]
```diff
@@ -74,6 +74,46 @@ rules:
...
+    - mobileclients
+  verbs:
+    - create
+    - delete
+    - get
+    - list
+    - patch
+    - update
+    - watch
+- apiGroups:
```
Comment (aliok) [inline deploy/role.yaml:92]: to give push and metrics related permissions to users, the operator itself needs to have permissions on those.

File: deploy/mobileclient_admin_role.yaml (hunk 1 above)
Comment (laurafitzgerald) [reply deploy/mobileclient_admin_role.yaml:31]: Can we include MSS one as well. I believe that it's group is: mobile-security-service.aerogear.com
and resources would be mobilesecurityserviceapps

File: deploy/mobileclient_admin_role.yaml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+      - push.aerogear.org
+    resources:
+      - androidvariants
+      - iosvariants
+      - pushapplications
+      - unifiedpushservers
```
Comment (laurafitzgerald) [inline deploy/mobileclient_admin_role.yaml]: Does MDC interact with the `unifiedpushservers`? I would have thought it only interacted with consumption crs?

File: deploy/mobileclient_admin_role.yaml [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+      - get
+      - list
+      - patch
+      - update
+      - watch
+  - apiGroups:
+      - metrics.aerogear.org
+    resources:
+      - appmetricsconfigs
+      - appmetricsservices
```
Comment (laurafitzgerald) [inline deploy/mobileclient_admin_role.yaml]: Same here with the `appmetricsservcies`?

**Comment by laurafitzgerald** [comment]
Some questions inline. Doing the verification now. :)

**Comment by jhellar** [comment]
@aliok can not pull the `aliok/mobile-developer-console-operator:jul10` image, got `Permission Denied`.

**Comment by aliok** [comment]
@jhellar mind trying `aliok/mobile-developer-console-operator:jul10-2` ?

File: deploy/mobileclient_admin_role.yaml (hunk 3 above)
Comment (aliok) [reply deploy/mobileclient_admin_role.yaml]: Good question.

**Comment by laurafitzgerald** [comment]
@aliok Is it `docker.io/aliok/mobile-developer-console-operator:jul10-2` we've been defaulting to quay recently. I tried `quay.io/aliok/mobile-developer-console-operator:jul10-2` first but I also couldn't pull the image.

**Comment by aliok** [comment]
@laurafitzgerald @jhellar yeah, it is Docker Hub, sorry

**Comment by laurafitzgerald** [comment]
Thanks @aliok. @jhellar if you are already verifying, will I leave you to it?

**Comment by jhellar** [comment]
@aliok ok, will try that, thanks. @laurafitzgerald yes, will verify it

**Review by jhellar — APPROVED** [review]
Verified.

**Comment by aliok** [comment]
Thanks @jhellar for finding the issues.
Merging this now.

**Pull request merged**